### PR TITLE
Implement method to add a document

### PR DESCRIPTION
**Pull request by siegy22**
Introduce new method `create_document` on the client which creates/uploads a document.

### CODE REVIEW

File: opengever/apiclient/client.py [hunk 1]
```diff
@@ -72,3 +74,16 @@ def listing(self, name, columns=[], raw=False, **kwargs):
 
     def update_object(self, **data):
         return self.session().patch(self.url, json=data).ok
+
+    @autowrap
+    def create_document(self, title, io, content_type, filename):
+        return self.session().post(self.url, json={
```
Comment (mbaechtold) [inline opengever/apiclient/client.py]: I'd appreciate a docstring explaining the parameter, especially the type of the parameter `io`.

File: opengever/apiclient/tests/test_client.py [hunk 2]
```diff
@@ -165,3 +167,20 @@ def test_documents_listing(self):
     def test_listing_passes_optional_params(self):
         listing = GEVERClient(url=self.dossier_url, username=self.regular_user).listing(name="documents", b_size=700)
         self.assertEqual(700, listing["b_size"])
+
+    def test_create_document(self):
+        document = GEVERClient(url=self.dossier_url, username=self.regular_user).create_document(
+            title='Ein Dokument',
+            fio=BytesIO(b'content of the document'),
```
Comment (mbaechtold) [inline opengever/apiclient/tests/test_client.py]: Shouldn't it be `io` instead of `fio`?

File: opengever/apiclient/client.py [hunk 3]
```diff
@@ -72,3 +74,16 @@ def listing(self, name, columns=[], raw=False, **kwargs):
 
     def update_object(self, **data):
         return self.session().patch(self.url, json=data).ok
+
+    @autowrap
+    def create_document(self, title, io, content_type, filename):
```
Comment (mbaechtold) [inline opengever/apiclient/client.py]: `io` will shadow an eventual import of the Python `io` module. Could we name it differently?

Comment (siegy22) [reply opengever/apiclient/client.py]: I named it `fio` but didn't stage that change in the client file, only in the test.

Comment (siegy22) [reply opengever/apiclient/client.py]: I guess `fio` isn't really better, what's the python convention on how to do that?

Comment (mbaechtold) [reply opengever/apiclient/client.py]: I don't think there is a Python convention. What about `content` or `data`?

Comment (siegy22) [reply opengever/apiclient/client.py]: `content` or `data` is the thing which comes out of the `io`. 

`ioish`? 😂

Comment (mbaechtold) [reply opengever/apiclient/client.py]: 🤔 Any input from @sebastianmanger or @lvonlanthen ?

File: opengever/apiclient/client.py (hunk 1 above)
Comment (mbaechtold) [reply opengever/apiclient/client.py]: Thanks!

File: opengever/apiclient/client.py [hunk 4]
```diff
@@ -72,3 +74,32 @@ def listing(self, name, columns=[], raw=False, **kwargs):
 
     def update_object(self, **data):
         return self.session().patch(self.url, json=data).ok
+
+    @autowrap
+    def create_document(self, title, file, content_type, filename):
+        """
+        :param str title: The title of the document
```
Comment (mbaechtold) [inline opengever/apiclient/client.py]:
```suggestion
        :param title: The title of the document
```

File: opengever/apiclient/client.py [hunk 5]
```diff
@@ -72,3 +74,32 @@ def listing(self, name, columns=[], raw=False, **kwargs):
...
+        :param fio: Readable IO which holds the content of the file
```
Comment (mbaechtold) [inline opengever/apiclient/client.py]:
```suggestion
        :param file: A file-like object which holds the content of the file
```

**Pull request merged**